modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItBuildIndexOneNodeTest.java

```diff
@@ -0,0 +1,334 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.apache.ignite.internal.catalog.CatalogManager;
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexDescriptor;
+import org.apache.ignite.internal.catalog.events.CatalogEvent;
+import org.apache.ignite.internal.catalog.events.MakeIndexAvailableEventParameters;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.sql.BaseSqlIntegrationTest;
+import org.apache.ignite.internal.sql.engine.util.QueryChecker;
+import org.apache.ignite.internal.table.distributed.replication.request.BuildIndexReplicaRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/** Integration test for testing the building of an index in a single node cluster. */
+public class ItBuildIndexOneNodeTest extends BaseSqlIntegrationTest {
+    private static final String TABLE_NAME = "TEST_TABLE";
+
+    private static final String ZONE_NAME = zoneName(TABLE_NAME);
+
+    private static final String INDEX_NAME = "TEST_INDEX";
+
+    @Override
+    protected int initialNodes() {
+        return 1;
+    }
+
+    @BeforeEach
+    void setup() {
+        setAwaitIndexAvailability(false);
+    }
+
+    @AfterEach
+    void tearDown() {
+        sql("DROP TABLE IF EXISTS " + TABLE_NAME);
+        sql("DROP ZONE IF EXISTS " + ZONE_NAME);
+
+        CLUSTER.runningNodes().forEach(IgniteImpl::stopDroppingMessages);
+    }
+
+    @Test
+    void testRecoveryBuildingIndex() throws Exception {
+        createZoneAndTable(ZONE_NAME, TABLE_NAME, 1, 1);
+
+        insertPersons(TABLE_NAME, new Person(0, "0", 10.0));
+
+        CompletableFuture<Void> awaitBuildIndexReplicaRequest = new CompletableFuture<>();
+
+        node().dropMessages((s, networkMessage) -> {
+            if (networkMessage instanceof BuildIndexReplicaRequest) {
+                awaitBuildIndexReplicaRequest.complete(null);
+
+                return true;
+            }
+
+            return false;
+        });
+
+        createIndex(TABLE_NAME, INDEX_NAME, "ID");
+
+        assertThat(awaitBuildIndexReplicaRequest, willCompleteSuccessfully());
+
+        assertFalse(isIndexAvailable(node(), INDEX_NAME));
+
+        // Let's restart the node.
+        CLUSTER.stopNode(0);
+        CLUSTER.startNode(0);
+
+        awaitIndexBecomeAvailable(node(), INDEX_NAME);
+    }
+
+    @Test
+    void testBuildingIndex() throws Exception {
+        createZoneAndTable(ZONE_NAME, TABLE_NAME, 1, 1);
+
+        insertPersons(TABLE_NAME, new Person(0, "0", 10.0));
+
+        createIndex(TABLE_NAME, INDEX_NAME, "ID");
+
+        awaitIndexBecomeAvailable(node(), INDEX_NAME);
```

Comment:
Shouldn't we also verify that a select would use the index and that it will find the person we just inserted?